ledger/db/src/test_utils/mock_ledger.rs

```diff
@@ -300,11 +301,139 @@ pub fn get_test_ledger_blocks(n_blocks: usize) -> Vec<(Block, BlockContents)> {
     blocks_and_contents
 }
 
+/// Get blocks with custom content in order to simulate conditions seen in
+/// production
+///
+/// * `outputs_per_recipient_per_block` - transaction outputs per account per
+///   block
+/// * `num_accounts` - number of accounts in the blocks
+/// * `num_blocks` - number of simulated blocks to create
+/// * `key_images_per_block` - number of key images per block
+/// * `max_token_id` - number of distinct token ids in blocks
+pub fn get_custom_test_ledger_blocks(
+    outputs_per_recipient_per_block: usize,
+    num_accounts: usize,
+    num_blocks: usize,
+    key_images_per_block: usize,
+    max_token_id: u64,
+) -> Vec<(Block, BlockContents)> {
+    let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+
+    // Number of total tx outputs in all blocks
+    let num_outputs: u64 =
+        (num_accounts * outputs_per_recipient_per_block * num_blocks * (max_token_id as usize + 1))
+            as u64;
+    assert!(num_outputs >= 16);
+
+    // Initialize other defaults
+    let picomob_per_output: u64 = (TOTAL_MOB / num_outputs) * 1_000_000_000_000;
+    let recipients = (0..num_accounts)
+        .map(|_| AccountKey::random(&mut rng).default_subaddress())
+        .collect::<Vec<_>>();
+    let block_version = BlockVersion::MAX;
+    let mut blocks_and_contents: Vec<(Block, BlockContents)> = Vec::new();
+    let mut previous_block: Option<Block> = None;
+
+    // Create the tx outs for all of the simulated blocks
+    for _ in 0..num_blocks as u64 {
```

Comment:
nit: is the `as u64` necessary, since it's just thrown away?

```suggestion
    for _ in 0..num_blocks {
```
```suggestion
    for _ in 0..num_blocks as u64 {
```